src/platforms/hubspot/client.ts

```diff
@@ -173,16 +173,23 @@ const makeClient = () => {
     schema: z.ZodSchema,
     properties?: string[],
   ): requestFunctionType<BatchReadObjectInput, TOutput> =>
-    request(({ ids }: BatchReadObjectInput) => ({
-      url: `/crm/${API_VERSION}/${module}/batch/read`,
-      method: 'POST',
-      json: {
-        properties: properties ?? null,
-        inputs: ids.map((id) => ({ id })),
-        propertiesWithHistory: null,
-      },
-      schema,
-    }));
+    request(
+      ({
+        ids,
+        after,
+        limit = HUBSPOT_MAX_PAGE_SIZE,
+      }: BatchReadObjectInput) => ({
+        url: `/crm/${API_VERSION}/${module}/batch/read`,
+        method: 'POST',
+        json: {
+          properties: properties ?? null,
```

Comment:
How will we fetch all properties this work when we add $native?

Reply:
Actually it looks like a POST request which might mean we might not run into a limit on the number of props

Reply:
Can you create a ticket to track adding $snative to batch gets?

Reply:
I actually haven't implemented it yet for the list endpoint either. Though that will probably happen within the unification where we fetch all properties and pass it through here. So allowing additional properties to be passed through might require a separate PR

Reply:
Sounds good, can we make sure to track all of this work in linear?